#VRML_SIM R2020a utf8
WorldInfo {
  gpsCoordinateSystem "WGS84"
}
Viewpoint {
  orientation 0.2627507562717918 0.940549404942068 0.21524139225930364 4.822253988164067
  position -6.234744843469182 3.1338466642378267 1.6239356251059178
  follow "SUGV:body"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 10 10
}
Solid {
  translation 1.78147 0.1 2.72486
  scale 2 2 2
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  boundingObject Box {
    size 0.1 0.1 0.1
  }
  physics Physics {
  }
  recognitionColors [
    0.9 0 0
  ]
}
WoodenBox {
  translation 2.66953 0.3 -1.11594
}
DEF SUGV Robot {
  translation 0.055341 0.16 0.753686
  rotation 0 1 0 -1.5707953071795862
  children [
    Camera {
      translation 0 0.04 0
      recognition Recognition {
      }
    }
    Compass {
      rotation 0 1 0 1.5708
    }
    GPS {
      rotation 0 1 0 1.57
    }
    DEF Wheel_RF HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.13 -0.0499998 -0.11
      }
      device [
        RotationalMotor {
          name "Motor_RF"
        }
      ]
      endPoint Solid {
        translation 0.13 -0.0499998 -0.11
        rotation 0 -1.1775671812919276e-16 1 1.5707999999999998
        children [
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 0 1
              }
            }
            geometry Cylinder {
              height 0.0254
              radius 0.0381
            }
          }
        ]
        name "Wheel_RF"
        boundingObject Cylinder {
          height 0.0254
          radius 0.0381
        }
        physics Physics {
        }
      }
    }
    DEF Wheel_RM HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.13 -0.05 0
      }
      endPoint Solid {
        translation 0.13 -0.05 0
        rotation 2.9439179532298196e-17 0 1 1.5707999999999989
        children [
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 1 0
              }
            }
            geometry Cylinder {
              height 0.0254
              radius 0.0381
            }
          }
        ]
        name "Wheel_RM"
        boundingObject Cylinder {
          height 0.0254
          radius 0.0381
        }
        physics Physics {
        }
      }
    }
    DEF Wheel_RB HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.13 -0.0500003 0.11
      }
      device [
        RotationalMotor {
          name "Motor_RB"
        }
      ]
      endPoint Solid {
        translation 0.13 -0.0500003 0.11
        rotation -7.850447875279521e-17 -7.850447875279521e-17 1 1.570799999999999
        children [
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 1 0
              }
            }
            geometry Cylinder {
              height 0.0254
              radius 0.0381
            }
          }
        ]
        name "Wheel_RB"
        boundingObject Cylinder {
          height 0.0254
          radius 0.0381
        }
        physics Physics {
        }
      }
    }
    DEF Wheel_LF HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.13 -0.0500008 -0.11
      }
      device [
        RotationalMotor {
          name "Motor_LF"
        }
      ]
      endPoint Solid {
        translation -0.13 -0.0500008 -0.11
        rotation -1.570089575055904e-16 -7.85044787527952e-17 1 1.570799999999999
        children [
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 0 1
              }
            }
            geometry Cylinder {
              height 0.0254
              radius 0.0381
            }
          }
        ]
        name "Wheel_LF"
        boundingObject Cylinder {
          height 0.0254
          radius 0.0381
        }
        physics Physics {
        }
      }
    }
    DEF Wheel_LM HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.13 -0.0500008 0
      }
      endPoint Solid {
        translation -0.13 -0.0500008 0
        rotation -1.9626119688198802e-17 0 1 1.570799999999999
        children [
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 1 0
              }
            }
            geometry Cylinder {
              height 0.0254
              radius 0.0381
            }
          }
        ]
        name "Wheel_LM"
        boundingObject Cylinder {
          height 0.0254
          radius 0.0381
        }
        physics Physics {
        }
      }
    }
    DEF Wheel_LB HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.13 -0.0500008 0.11
      }
      device [
        RotationalMotor {
          name "Motor_LB"
        }
      ]
      endPoint Solid {
        translation -0.13 -0.0500008 0.11
        rotation -1.9626119688198792e-16 0 1 1.5707999999999995
        children [
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 1 0
              }
            }
            geometry Cylinder {
              height 0.0254
              radius 0.0381
            }
          }
        ]
        name "Wheel_LB"
        boundingObject Cylinder {
          height 0.0254
          radius 0.0381
        }
        physics Physics {
        }
      }
    }
    DEF Body Solid {
      translation 0 0.0178 0
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 1 0 0
            }
          }
          geometry Box {
            size 0.2286 0.16 0.3048
          }
        }
      ]
      name "body"
      boundingObject Box {
        size 0.2286 0.16 0.3048
      }
      physics Physics {
      }
    }
  ]
  name "SUGV"
  boundingObject Transform {
    translation 0 0.0178 0
    children [
      Box {
        size 0.2286 0.16 0.3048
      }
    ]
  }
  physics Physics {
  }
  recognitionColors [
    0 0 0
  ]
  controller "NGCP_2020_Robot_Controll_Camera_Rec"
}
